WordPress/src/main/java/org/wordpress/android/util/encryption/EncryptionUtils.java

```diff
@@ -43,16 +54,21 @@ public static String generateJSONEncryptedLogs(final byte[] publicKey,
                                                    final List<String> logMessages) throws JSONException {
         // Schema version
         JSONObject encryptionDataJson = new JSONObject();
-        encryptionDataJson.put("keyedWith", KEYED_WITH);
+        encryptionDataJson.put(FIELD_KEYED_WITH, KEYED_WITH);
+
+        // Generate Logs UUID
+        String source = KEYED_WITH + ANDROID_PLATFORM + System.currentTimeMillis();
+        byte[] bytes = source.getBytes();
+        encryptionDataJson.put(FIELD_LOGS_ID, UUID.nameUUIDFromBytes(bytes));
```

Comment:
This looks like it produces a solid UUIDv3, and it's exceedingly unlikely we'd ever call this method twice in the same millisecond. Out of curiosity though, why not use a UUIDv4 and avoid the concatenation and byte wrangling?

Reply:
Good question! I was reading about V3 and V4 and I ended up in this blog post where they explain what are the chances of creating unique codes https://towardsdatascience.com/are-uuids-really-unique-57eb80fc2a87

They mention it is quite good for practical purposes, which is true, but since we are potentially using this logic in other apps and platforms I thought it was a good idea to have more control over it and remove the tiny bit of chance of getting a duplicate.

Does it make sense? :)